### ISSUE TITLE
[Feature Request] API - Multi-disc response for "album"

### ISSUE THREAD

**Issue by mitchray** [ampache/ampache#2776]
I'm not sure if this is a feature request or bug report... currently the album/advanced_search only returns the first disc of a multi-disc album

Example call ```/server/json.server.php?action=album&filter=169```

Currently testing on source-changes but appears to be on 4.3 also

For context the legacy GUI ```/albums.php?action=show&album=169``` pulls in disc 1 (album ID 169) and disc 2 (album ID 170) (same MBID?)

![firefox_kgN1D1oP2o](https://user-images.githubusercontent.com/5735900/108699277-251b0900-7559-11eb-855f-89fa4e3460ec.png)

Album/disc 1 response:
```
{
    "album": [
        {
            "id": "169",
            "name": "Dead Set on Living",
            "artist": {
                "id": "130",
                "name": "Cancer Bats"
            },
            "time": 0,
            "year": 2012,
            "tracks": [],
            "songcount": 20,
            "disk": 2,
            "genre": [
                {
                    "id": "33",
                    "name": "Hardcore Punk"
                }
            ],
            "art": "http:\/\/192.168.1.111\/ampache-preview\/image.php?object_id=169&object_type=album&auth=auth",
            "flag": 0,
            "preciserating": null,
            "rating": null,
            "averagerating": null,
            "mbid": "77d2bbf4-aa41-4013-97e7-e5627170f2b7"
        }
    ]
}
```

Disc 2 response:
```
{
    "album": [
        {
            "id": "170",
            "name": "Dead Set on Living",
            "artist": {
                "id": "130",
                "name": "Cancer Bats"
            },
            "time": 0,
            "year": 2012,
            "tracks": [],
            "songcount": 20,
            "disk": 2,
            "genre": [
                {
                    "id": "33",
                    "name": "Hardcore Punk"
                }
            ],
            "art": "http:\/\/192.168.1.111\/ampache-preview\/image.php?object_id=170&object_type=album&auth=auth",
            "flag": 0,
            "preciserating": null,
            "rating": null,
            "averagerating": null,
            "mbid": "77d2bbf4-aa41-4013-97e7-e5627170f2b7"
        }
    ]
}
```

You can see that both albums report the combined song count so its recognising both somewhere in the background, also both are disk 2

**Comment by lachlan-00** [comment]
Something like include_suite to include all the IDs.

Current behaviour is to group by the min Id.

**Comment by lachlan-00** [comment]
housekeeping Q.

if we include all disks in grouping does that count that towards a limit?
![image](https://user-images.githubusercontent.com/1305249/108795160-ed9c7300-75d1-11eb-8f09-9c505fbeb4da.png)

e.g include a limit of 50 but return more disks (say some doubles) which means we return 62 objects

OR

do we split that into disks/id's inside the object into a disk grouping instead?

```
{
    "album": [
        {
            "id": "170",
            "name": "Dead Set on Living",
            "artist": {
                "id": "130",
                "name": "Cancer Bats"
            },
            "time": 0,
            "year": 2012,
            "tracks": [],
            "songcount": 20,
            "disk": 2,
            "DISKGROUPorWHATEVER": [
                {
                    "id": "129",
                    "disk": "1"
                },
                {
                    "id": "130",
                    "name": "2"
                }
            ],
            "genre": [
                {
                    "id": "33",
                    "name": "Hardcore Punk"
                }
            ],
            "art": "http:\/\/192.168.1.111\/ampache-preview\/image.php?object_id=170&object_type=album&auth=auth",
            "flag": 0,
            "preciserating": null,
            "rating": null,
            "averagerating": null,
            "mbid": "77d2bbf4-aa41-4013-97e7-e5627170f2b7"
        }
    ]
}
```

I kind of like extending the album object into disk id groups instead as it will allow you to always return the 'album' with a clear disk/id list of it's contents

**Comment by lachlan-00** [comment]
@AshotN when you have time this is what i'm thinking.

I propose making disk into diskcount and adding albumgroup (or a better name) which holds the disk ids.
Then for things like album_songs we allow selecting a single disk (optional)

album_suite is created on construct so the data is already there and it will maintain the grouping and keep albums in a single item

```
{
    "album": [
        {
            "id": "15697",
            "name": "Mellon Collie and the Infinite Sadness",
            "artist": {
                "id": "1240",
                "name": "Smashing Pumpkins"
            },
            "time": 0,
            "year": 1995,
            "tracks": [],
            "songcount": 28,
            "disk": 2,
            "albumgroup": [
                {
                    "id": 15698,
                    "disk": 1
                },
                {
                    "id": 15697,
                    "disk": 2
                }
            ],
            "genre": [
                {
                    "id": "33",
                    "name": "Alternative Rock"
                }
            ],
            "art": "https:\/\/music.com.au\/image.php?object_id=15697&object_type=album&auth=eeb9f1b6056246a7d563f479f518bb34",
            "flag": 0,
            "preciserating": null,
            "rating": null,
            "averagerating": null,
            "mbid": "5bc55bab-b2be-4566-8983-00108b1d5d8f"
        }
    ]
}
```

```
<?xml version="1.0" encoding="UTF-8" ?>
<root>
<total_count>31681</total_count>
<album id="15697">
	<name><![CDATA[Mellon Collie and the Infinite Sadness]]></name>
	<artist id="1240"><![CDATA[Smashing Pumpkins]]></artist>
	<time>0</time>
	<year>1995</year>
	<tracks></tracks>
	<songcount>28</songcount>
	<disk>2</disk>
	<albumgroup>
		<album id="15698"><disk>1<disk></album>
		<album id="15697"><disk>2<disk></album>
	</albumgroup>
	<genre id="33"><![CDATA[Alternative Rock]]></genre>
	<art><![CDATA[https://music.com.au/image.php?object_id=15697&object_type=album&auth=eeb9f1b6056246a7d563f479f518bb34]]></art>
	<flag>0</flag>
	<preciserating></preciserating>
	<rating></rating>
	<averagerating></averagerating>
	<mbid><![CDATA[5bc55bab-b2be-4566-8983-00108b1d5d8f]]></mbid>
</album>
</root>
```

**Comment by mitchray** [comment]
Also for consideration, what would happen if ```&include=songs"``` is present, would each <albumgroup> have its own <tracks> or would there be a disc ID added for each track?

**Comment by lachlan-00** [comment]
in api5 the tracks would be part of the tracks part above (if included)

**Comment by lachlan-00** [comment]
> in api5 the tracks would be part of the tracks part above (if included)

but this is broken right now in json so i'll fix

**Comment by lachlan-00** [comment]
i've fixed up all the data now. song disk wasn't being included in responses at all which is odd. i'm leaning away from including disks as sub items now that it's actually returning data.

I need to do more examples with inclusions but i'm out of time today.

@usox i can't work out (no time) this repository code, it's not including anything in my testing (pasted json but it's xml as well) it's also marked as depreciated so not sure what's up the response is empty ``"tracks": [],```` but should be filled with my includes

getByArtist and getByAlbum
```
            // Handle includes
            $albums = (in_array("albums", $include))
                ? self::albums(static::getAlbumRepository()->getByArtist($artist), array(), $user_id, false)
                : array();
            $songs = (in_array("songs", $include))
                ? self::songs(static::getSongRepository()->getByArtist($artist), $user_id, false)
                : array();
```

here is a master example (working)
https://github.com/ampache/python3-ampache/blob/api4/docs/json-responses/get_indexes%20(album%20with%20include).json

source-changes (broken)
https://github.com/ampache/python3-ampache/blob/master/docs/json-responses/get_indexes%20(album%20with%20include).json

**Comment by usox** [comment]
Uhmm I'd say, it's broken because b62f73911498379b87974674ecec2c6db13e1eeb from master is missing in source-changes... ?!

**Comment by lachlan-00** [comment]
it is, that's fixed now just working on the new responses

**Comment by lachlan-00** [comment]
Playlist ratings were ignored because of the query for playlists. I need to fix that query up again as it's not returning what I want

**Comment by lachlan-00** [comment]
when including tracks you'll get the disk entry as well.
https://github.com/ampache/python3-ampache/blob/master/docs/json-responses/album%20(with%20include).json

same for song
https://github.com/ampache/python3-ampache/blob/master/docs/json-responses/song.json

I don't think all the id's need to be included for albums. With grouping on they aren't used and with grouping disabled the tracks aren't included in the response. they exist but can be ignored as all the queries will return a single id.

i'm still going to change disk to diskCount though.

**Comment by mitchray** [comment]
I haven't tested but I think this is resolved from #3374
